react/components.md

```diff
@@ -0,0 +1,98 @@
+### Introduction
+
+In this lesson we'll be going over the basics of React components - what they do, and how to write them. Make sure to use the project you set up in the [Setting Up A React Environment lesson](https://github.com/TheOdinProject/top-meta/issues/221), but try not to copy and paste any code while you're coding along. 
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*   What are components?
+*   How are components created?
+*   Where do components live?
+
+### What Are Components
+
+The beauty of React is that it allows you to break a UI (User Interface) down into independent reusable chunks, which we will refer to as components. The following picture should give you an idea of how to do that when building a very basic app. 
+
+![Component Example](./imgs/00.png)
+
+For example, this simple website could be broken into the following components:
+
+- `App`, which represents your main application and will be the parent of all other components.
+- `Navbar`, which will be the navigation bar.
+- `MainArticle`, which will be the component that renders your main content.
+- `NewsletterForm`, which is a simple form that lets a user input their email to receive the weekly newsletter.
+
+Think of these reusable chunks as JavaScript functions which can take some kind of input and return a React element.
+
+### How To Create Components
+
+To get the feel of working with components, we're going to practice creating functional components. What are functional components? Javascript functions! Is it really that simple? Let's have a look.
+
+~~~jsx
+function Greeting() {
+  return <h1>"I swear by my pretty floral bonnet, I will end you."</h1>;
+}
+~~~
+
+This might look mostly familiar to you - it's a simple JavaScript function, which returns JSX. Open up the project you were working on, create a new file named `Greeting.jsx`, and in that file write your own handmade functional component. Name it whatever you wish, have it return whatever JSX you wish. Are you done? Check the naming of your function! Is it capitalized? Keep this key difference in mind, components that aren't capitalized might not run as expected - which is why we capitalized `Greeting()`.
```

Comment:
I think breaking this into two paragraph fits.
```suggestion
This might look mostly familiar to you - it's a simple JavaScript function, which returns JSX. Open up the project you were working on, create a new file named `Greeting.jsx`, and in that file write your own handmade functional component. Name it whatever you wish, have it return whatever JSX you wish. 

Are you done? Check the naming of your function! Is it capitalized? Keep this key difference in mind, components that aren't capitalized might not run as expected - which is why we capitalized `Greeting()`.
```